python/cudf/cudf/_libxx/copying.pyx

```diff
@@ -34,3 +41,25 @@ def gather(Table source_table, Column gather_map):
         column_names=source_table._column_names,
         index_names=source_table._index._column_names
     )
+
+
+def shift(Column input, int offset, Scalar fill_value=None):
+
+    if not isinstance(fill_value, Scalar):
+        fill_value = Scalar(fill_value, input.dtype)
```

Comment:
```suggestion
        cdef Scalar fill_value = Scalar(fill_value, input.dtype)
```

Reply:
What's the significance of this? Cython optimization?

Reply:
If you don't define this as a `cdef` then you can't access the `cdef` functions of the `Scalar` class. My change above defines `fill_value` as a generic `object` since someone would presumably pass down something like a Python integer or whatnot, so Cython isn't smart enough to know that `fill_value` is a `Scalar` that it can call cpp functions against.